api/api.go

```diff
@@ -234,12 +243,27 @@ func procGetNetwork(c libnetwork.NetworkController, vars map[string]string, body
 func procGetNetworks(c libnetwork.NetworkController, vars map[string]string, body []byte) (interface{}, *responseStatus) {
 	var list []*networkResource
 
-	// If query parameter is specified, return a filtered collection
-	if name, queryByName := vars[urlNwName]; queryByName {
-		nw, errRsp := findNetwork(c, name, byName)
-		if errRsp.isOK() {
+	// Look for query filters and validate
+	name, queryByName := vars[urlNwName]
+	shortID, queryByPid := vars[urlNwPID]
+	if queryByName && queryByPid {
+		// vars are populated from the URL, following cannot happen
+		panic("Unsupported query composition URL variable parameter for endpoint")
+	}
```

Comment:
Even though this cannot happen (unless a bug in the mux), I will replace the panic() with a Forbidden response message.